Modules/Core/Common/include/itkImageSource.hxx

```diff
@@ -39,11 +39,8 @@ namespace itk
 template <typename TOutputImage>
 ImageSource<TOutputImage>::ImageSource()
 {
-  // Create the output. We use static_cast<> here because we know the default
-  // output must be of type TOutputImage
-  typename TOutputImage::Pointer output = static_cast<TOutputImage *>(this->MakeOutput(0).GetPointer());
   this->ProcessObject::SetNumberOfRequiredOutputs(1);
-  this->ProcessObject::SetNthOutput(0, output.GetPointer());
+  this->ProcessObject::SetNthOutput(0, TOutputImage::New().GetPointer());
```

Comment:
Please add comment like: "Generally MakeOutput(n) should be used, but because a virtual method should not be called in a constructor the output is directly constructed."

It is important to note how things should be done, and why they are not done that way for this case. There have been bugs before related to not using MakeOutput.

Reply:
Thanks for the suggestion @blowekamp

> There have been bugs before related to not using MakeOutput.

Can you possibly still recall what such a bug looked like? Just for my understanding!

Reply:
Don't recall the specifics... There are cases when a Process object can have it's outputs removed, and then during execution it needs to make them again.

Maybe f0d570ecfce3d7ca2f099f3f4e18c9e343fa4fc6?

Reply:
Interesting! In general, I think `this->MakeOutput(i)` calls are not necessary, inside a constructor of a class, when the `MakeOutput(i)` override of this class unconditionally just does the same `OutputType::New()` call for any index value `i`.

Reply:
I agree they are not necessary. I am just suggestion adding documentation as to why the canonical way of creating a new output is not being used.

Reply:
I think I understand your point, thanks! I would just like to get clearer in which particular cases `this->MakeOutput(i)` would actually be useful, in a constructor.

Is `x->MakeOutput(i)` also intended to be called from outside? I mean, would an ITK _end_ user actually call `x->MakeOutput(i)`?

Reply:
Hrmm... This might undo fixes made here: https://github.com/InsightSoftwareConsortium/ITK/pull/961

Reply:
Sorry, what do you mean? Your pull request #961 (merged on May 30, 2019) was about different source files (itkDataObject.h, itkProcessObject.cxx, itkDataObjectAndProcessObjectTest.cxx). Whereas this PR (PR #4654) only does internal style changes to the constructors of `ImageSource`, `MeshSource`, and `ImageToMeshFilter`. I think this pull request is innocent  😇

Reply:
Oh, sorry Bradley, what do you want there? Something like

    // Basically equivalent to SetNthOutput(0, MakeOutput(0)).
    this->ProcessObject::SetNthOutput(0, TOutputImage::New().GetPointer())

?

I hope it does not need to be longer than one line of comment. This pull request is intended to simplify the code. The idea is: Just call `TOutput::New()` if that does the job!

Reply:
What do you mean "As the idea would be to use New(), rather that MakeOutput, in multiple places in the source tree"?

Reply:
Well, this pull request already proposes to replace three `MakeOutput` calls with `TOutput::New()`. I don't think it's nice to have an extensive multi-line comment copy-and-pasted multiple times.